custom_components/iotawatt/sensor.py

```diff
@@ -1,137 +1,266 @@
     DEVICE_CLASS_POWER,
+    DEVICE_CLASS_POWER_FACTOR,
     DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_CURRENT_AMPERE,
     ELECTRIC_POTENTIAL_VOLT,
     ENERGY_WATT_HOUR,
+    FREQUENCY_HERTZ,
+    PERCENTAGE,
+    POWER_VOLT_AMPERE,
     POWER_WATT,
 )
 from homeassistant.core import callback
-from homeassistant.helpers import entity_registry
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers import entity, entity_registry, update_coordinator
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.restore_state import RestoreEntity
 from homeassistant.util import dt
 
-from . import IotaWattEntity
-from .const import COORDINATOR, DOMAIN, SIGNAL_ADD_DEVICE
+from .const import (
+    ATTR_LAST_UPDATE,
+    DOMAIN,
+    VOLT_AMPERE_REACTIVE,
+    VOLT_AMPERE_REACTIVE_HOURS,
+)
+from .coordinator import IotawattUpdater
 
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
-    """Add sensors for passed config_entry in HA."""
+@dataclass
+class IotaWattSensorEntityDescription(SensorEntityDescription):
+    """Class describing IotaWatt sensor entities."""
 
-    coordinator = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
-    entities = []
+    value: Callable | None = None
 
-    for idx, ent in enumerate(coordinator.data["sensors"]):
-        entity = IotaWattSensor(
-            coordinator=coordinator,
-            entity=ent,
-            mac_address=coordinator.data["sensors"][ent].hub_mac_address,
-            name=coordinator.data["sensors"][ent].getName(),
-        )
-        entities.append(entity)
 
-    async_add_entities(entities)
+ENTITY_DESCRIPTION_KEY_MAP: dict[str, IotaWattSensorEntityDescription] = {
+    "Amps": IotaWattSensorEntityDescription(
+        "Amps",
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        device_class=DEVICE_CLASS_CURRENT,
+        entity_registry_enabled_default=False,
+    ),
+    "Hz": IotaWattSensorEntityDescription(
+        "Hz",
+        native_unit_of_measurement=FREQUENCY_HERTZ,
+        state_class=STATE_CLASS_MEASUREMENT,
+        icon="mdi:flash",
+        entity_registry_enabled_default=False,
+    ),
+    "PF": IotaWattSensorEntityDescription(
+        "PF",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=STATE_CLASS_MEASUREMENT,
+        device_class=DEVICE_CLASS_POWER_FACTOR,
+        value=lambda value: value * 100,
+        entity_registry_enabled_default=False,
+    ),
+    "Watts": IotaWattSensorEntityDescription(
+        "Watts",
+        native_unit_of_measurement=POWER_WATT,
+        state_class=STATE_CLASS_MEASUREMENT,
+        device_class=DEVICE_CLASS_POWER,
+    ),
+    "WattHours": IotaWattSensorEntityDescription(
+        "WattHours",
+        native_unit_of_measurement=ENERGY_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
+    ),
```

Comment:
```suggestion
    "WattHours": IotaWattSensorEntityDescription(
        "WattHours",
        native_unit_of_measurement=ENERGY_WATT_HOUR,
        state_class=STATE_CLASS_TOTAL_INCREASING,
        device_class=DEVICE_CLASS_ENERGY,
    ),
```

Reply:
The class is already set on the high resolution sensor.